wagtail_streamfield_migration_toolkit/migrate_operation.py

```diff
@@ -126,22 +116,13 @@ def migrate_stream_data_forward(self, apps, schema_editor):
             model.objects.bulk_update(updated_model_instances_buffer, [self.field_name])
 
         # For models without revisions
-        if not has_revisions:
+        if not revision_query_maker.has_revisions:
             return
 
-        # TODO support for wagtail 3 ?
-        ContentType = apps.get_model("contenttypes", "ContentType")
-        contenttype_id = ContentType.objects.get_for_model(model).id
-
-        if self.revisions_from is not None:
-            revision_query = Q(
-                created_at__gte=self.revisions_from,
-                content_type_id=contenttype_id,
-            ) | Q(id__in=live_and_latest_revision_ids)
-            # we always update latest and live revision if available
-        else:
-            revision_query = Q(content_type_id=contenttype_id)
-        revision_queryset = Revision.objects.filter(revision_query)
+        revision_query = revision_query_maker.make_revision_query()
+        revision_queryset = revision_query_maker.RevisionModel.objects.filter(
```

Comment:
Could we get the `revision_query_maker` just to return the queryset? Getting its `RevisionModel` and then using it with its `query` feels like something that requires a bit too much internal knowledge of the class - it would be clearer just to return a queryset I think

Reply:
Would it be fine to get both the `RevisionModel` and the revision query/or queryset then? Since I will need access to the `RevisionModel` for the `bulk_update` further down.

Reply:
I think that's okay. (or you could give it a bulk_update method directly, either works)